tests/cpp/test_multidevice_communications.cpp

```diff
@@ -105,7 +108,10 @@ TEST_P(CommunicationTest, Allgather) {
   FusionGuard fg(&container);
   auto* in = makeContigTensor(2);
   in->setDeviceMesh(full_mesh_);
+  in->axis(0)->parallelize(ParallelType::DIDx);
```

Comment:
This is needed since `postAllGather` now relies on parallelization annotation of tensorviews to get the sharded logical axis.